src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.NativeAot.cs

```diff
@@ -18,6 +18,16 @@ public static partial class Marshal
         internal static int SizeOfHelper(Type t, bool throwIfNotMarshalable)
         {
             Debug.Assert(throwIfNotMarshalable);
+
+            if (t is not RuntimeType)
+                throw new ArgumentException(SR.Argument_MustBeRuntimeType);
+
+            if (t.IsPointer /* or IsFunctionPointer */)
+                return IntPtr.Size;
+
+            if (t.IsByRef || t.IsGenericTypeDefinition || t.IsArray || t.ContainsGenericParameters)
```

Comment:
```suggestion
            if (t.IsByRef || t.IsArray || t.ContainsGenericParameters)
```
Generic type definitions should be covered by ContainsGenericParameters condition